src/components/signup/EmailForm.js

```diff
@@ -39,31 +41,38 @@ class EmailForm extends React.Component<Props, State> {
   }
 
   handleSubmit = () => {
-    if (this.isValid) {
+    const isValid = this.checkErrors()
```

Comment:
isValid will always be truthy, as it's a Promise object.

You need to "await" for the response.

Reply:
done